docs/targets/solana.rst

```diff
@@ -1,34 +1,95 @@
-- Solana uses different accounts for the program code, and the contract data.
-- Programs are upgradable. There is no need to implement upgrades in Solidity.
+Solana Overview
+_______________
+
+As the underlying Solana environment is different than that of Ethereum, Solidity inner workings
+have been modified to function properly. For example, A Solidity contract on Solana utilizes two accounts: a
+data account and a program account. The program account stores the contract's executable binary and owns
+the data account, which holds all the storage variables. On Ethereum a single account can store executable
+code and data.
+
+Contract upgrades
++++++++++++++++++
+
+Provided that the data layout from a new contract is compatible with that of an old one, it is possible
+to update the binary in the program account and retain the same data, rendering
+contract upgrades implemented in Solidity unnecessary. Solana's CLI tool provides a command to both do
+an initial deploy of a program, and redeploy it later.:
+
+.. code-block:: bash
+
+    solana program deploy --program-id <KEYPAIR_FILEPATH> <PROGRAM_FILEPATH>
+
+where ``<KEYPAIR_FILEPATH>`` is the program's keypair json file and ``<PROGRAM_FILEPATH>``
+is the program binary ``.so`` file. For more information about redeploying a program,
+check `Solana's documentation <https://docs.solana.com/cli/deploy-a-program#redeploy-a-program>`_.
+
+Data types
+++++++++++
+
+- An account address consists of a 32-bytes key, which is represented by the ``address`` type. This data model
+  differs from Ethereum 20-bytes addresses.
+- Solana's virtual machine registers are 64-bit wide, so 64-bit integers ``uint64`` and ``int64`` are preferable over
+  ``uint256`` and ``int256``. Likewise, all balances and values on Solana are 64-bit wide, so the builtin functions for
+  *address* ``.balance``, ``.transfer()`` and ``.send()`` use 64-bit integers.
+- An address literal has to be specified using the ``address"36VtvSbE6jVGGQytYWSaDPG7uZphaxEjpJHUUpuUbq4D"`` syntax.
+- Ethereum syntax for addresses ``0xE0f5206BBD039e7b0592d8918820024e2a7437b9`` is not supported.
+
+Runtime
++++++++
+
+- The Solana target requires `Solana <https://www.solana.com/>`_ v1.8.1.
+- Function selectors are eight bytes wide and known as *discriminators*.
 - Solana provides different builtins, e.g. ``tx.program_id`` and ``tx.accounts``.
-- Function selectors are eight bytes and known as *discriminators*.
 
-This is how to build your Solidity for Solana:
+Compute budget
+++++++++++++++
+
+On Ethereum, when calling a smart contract function, one needs to specify the amount of gas the operation is allowed
+to use. Gas serves to pay for a contract execution on chain and can be a way for giving a contract priority execution
+when extra gas is offered in a transaction. Each EVM instruction has an associated gas value, which translates to real
+ETH cost. As there is no limit gas for the amount of gas one can provide in a transaction,
```

Comment:
> As there is no limit gas for the amount of gas one can provide in a transaction,

Ethereum has a gas limit field in a transaction for exactly this

Reply:
What does this mean?

Reply:
There is no limit for what you can input int the `gas limit` field.

Reply:
Can you elaborate more on what you are trying to point here?

Reply:
> As there is no limit gas for the amount of gas one can provide in a transaction,

This is not clear. This could be read as: 
- it is not possible to limit the amount of gas that a transaction can use.
- you can specify any value you want as a gas limit (you're limited by your balance after all)

Also: do you mean `no gas limit`?

Reply:
> Also: do you mean `no gas limit`?

`limit gas` was your suggested wording.

`there is no limit gas except for the amount of eth provided in the transaction for gas. `

Reply:
Fixed.